hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,393 @@
+        if sample_group is not None:
+            for key, value in sample_group.attrs.items():
+                emd.sample[key] = value
+        # Extract comments:
+        comments_group = emd_file.get('comments')
+        if comments_group is not None:
+            for key, value in comments_group.attrs.items():
+                emd.comments[key] = value
+        # Extract signals:
+        node_list = list(emd_file.keys())
+        for key in ['user', 'microscope', 'sample', 'comments']:  # Nodes which are not the data!
+            if key in node_list:
+                node_list.pop(node_list.index(key))  # Pop all unwanted nodes!
+        # One node should remain, the data node (named 'data', 'signals', 'experiments', ...)!
+        assert len(node_list) == 1, 'Dataset location is ambiguous!'
+        data_group = emd_file.get(node_list[0])
+        if data_group is not None:
+            for name, group in data_group.items():
+                if isinstance(group, h5py.Group):
+                    if group.attrs.get('emd_group_type') == 1:
+                        emd._read_signal_from_group(name, group, load_to_memory)
+        # Close file and return EMD object:
+        if load_to_memory:
+            emd_file.close()
+        return emd
+
+    def save_to_emd(self, filename='datacollection.emd'):
+        """Save :class:`~.EMD` data in a file with emd(hdf5)-format.
+
+        Parameters
+        ----------
+        filename : string, optional
+            The name of the emd-file in which to store the signals.
+            The default is 'datacollection.emd'.
+
+        Returns
+        -------
+        None
+
+        """
+        self._log.debug('Calling save_to_emd')
+        # Open file:
+        emd_file = h5py.File(filename, 'w')
+        # Write version:
+        ver_maj, ver_min = EMD_VERSION.split('.')
+        emd_file.attrs['version_major'] = ver_maj
+        emd_file.attrs['version_minor'] = ver_min
+        # Write user:
+        user_group = emd_file.require_group('user')
+        for key, value in self.user.items():
+            user_group.attrs[key] = value
+        # Write microscope:
+        microscope_group = emd_file.require_group('microscope')
+        for key, value in self.microscope.items():
+            microscope_group.attrs[key] = value
+        # Write sample:
+        sample_group = emd_file.require_group('sample')
+        for key, value in self.sample.items():
+            sample_group.attrs[key] = value
+        # Write comments:
+        comments_group = emd_file.require_group('comments')
+        for key, value in self.comments.items():
+            comments_group.attrs[key] = value
+        # Write signals:
+        signal_group = emd_file.require_group('signals')
+        for signal in self.signals.values():
+            self._write_signal_to_group(signal_group, signal)
+        # Close file and return EMD object:
+        emd_file.close()
+
+    def print_info(self):

```

Comment:
We've changed the printing to just logging (to leave as few as possible printing statements in general). Could you adapt this function as well? [here](https://github.com/hyperspy/hyperspy/blob/ea59b223c5e1301c097d0b24d126e93c003d4ae0/hyperspy/io_plugins/semper_unf.py#L631) is an example